src/js/redux_state/modules/single-sign-on/access-request.js

```diff
@@ -1,188 +1,203 @@
 import Immutable from 'immutable'
 import {makeActions} from '../'
 import router from '../router'
+import { actions as simpleDialogActions } from '../simple-dialog'
 
 export const actions = makeActions('single-sign-on/access-request', {
-  checkUserLoggedIn: {
+  setInfoComplete: {
+    expectedParams: []
+  },
+  getDid: {
     expectedParams: [],
-    creator: (params) => {
-      return (dispatch, getState, {services}) => {
-        const user = services.auth.currentUser
-        const path = getState().toJS().singleSignOn.accessRequest.entity.path
-        if (user == null) {
-          dispatch(router.pushRoute({
-            pathname: '/login',
-            query: {
-              callbackUrl: path
-            }
-          }))
+    async: true,
+    creator: () => {
+      return async (dispatch, getState, {services}) => {
+        const did = await services.storage.getItem('did')
+        if (did) {
+          dispatch(actions.setDid({did}))
         }
       }
     }
   },
-  requestedDetails: {
-    expectedParams: ['details'],
+  confirmAccess: {
```

Comment:
This action is a bit too long and it's not clear at a first glance what it's doing - I would suggest extracting methods that would nicely describe what it's doing, for example: 
```
getClaims(key)
getDecryptedWIF()
getToken()
postToken()
```